### PR TITLE
Initial commit, first draft

### PR DESCRIPTION
**Pull request by EttoreFoti**
Waiting for approval on first draft of definition, composition and claim for crateri-aws-stack deploy

### CODE REVIEW

File: claim.yaml [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+kind: aws-stack
+metadata:
+  name: krateo-module-core
+spec:
+  compositionSelector:
+    matchLabels:
+      platform: aws
+  eks:
+    name: #ex clusterRef
+    component_id: #values.component_id
```
Comment (braghettos) [inline claim.yaml:12]: Rename to clusterName

File: composition.yaml [hunk 2]
```diff
@@ -0,0 +1,418 @@
...
+              endpointPrivateAccess: true
+              endpointPublicAccess: true
+            writeConnectionSecretToRef:
+              namesapce: crossplane-system
+            providerConfigRef:
+              name: aws-stack-template-provider        
+      patches:
+        - fromFieldPath: "spec.eks.name"
+          toFieldPath: "metadata.name"
+        - fromFieldPath: "spec.eks.resourcesVpcConfig.subnetIdRefs"
```
Comment (braghettos) [inline composition.yaml:29]: I don't understand why we are not taking as a fromFieldPath the following value:

'spec.network.name'

File: composition.yaml [hunk 3]
```diff
@@ -0,0 +1,418 @@
...
+          toFieldPath: "spec.forProvider.resourcesVpcConfig.subnetIdRefs[0]"
```
Comment (braghettos) [inline composition.yaml:30]: I would specify better the field:

spec.forProvider.resourcesVpcConfig.subnetIdRefs[0].name

File: composition.yaml [hunk 4]
```diff
@@ -0,0 +1,418 @@
...
+            - type: string
+              string:
+                fmt: "%s-sn0" 
+        - fromFieldPath: "spec.network.name"
+          toFieldPath: "spec.forProvider.resourcesVpcConfig.subnetIdRefs[1]" 
+          transforms:
+            - type: string
+              string:
+                fmt: "%s-sn1"
+        - fromFieldPath: "sspec.eks.component_id"
```
Comment (braghettos) [inline composition.yaml:41]: sspec is a typo

**Comment by EttoreFoti** [comment]
In the last commit, I should have fixed all the points you pointed

File: composition.yaml (hunk 4 above)
Comment (braghettos) [reply composition.yaml]: fromFieldPath should be spec.name

File: definition.yaml [hunk 5]
```diff
@@ -0,0 +1,263 @@
...
+                          type: string
+                        region:
+                          type: string
+                        cidrBlock:
+                          type: string
+                        enableDnsSupport:
+                          type: boolean
+                        enableDnsHostNames:
+                          type: boolean
+                        providerConfigRef:
```
Comment (braghettos) [inline definition.yaml:69]: Useless since it will be determined within the composition

File: definition.yaml [hunk 6]
```diff
@@ -0,0 +1,263 @@
...
+                          type: string
+                        mapPublicIPOnLaunch:
+                          type: boolean
+                        cidrBlock:
+                          type: string
+                        availabilityZone:
+                          type: string
+                        vpcIdRef: #.name
+                          type: string
+                        providerConfigRef: #.name
```
Comment (braghettos) [inline definition.yaml:104]: Useless since it will be determined within the composition

File: definition.yaml [hunk 7]
```diff
@@ -0,0 +1,263 @@
...
+                        instanceType:
+                          type: string
+                    subnetRefs:
+                      type: object
+                      properties:
+                        name:
+                          type: strings
+                    nodeRoleRef: #.name
+                      type: string
+                    providerConfigRef: #.name
```
Comment (braghettos) [inline definition.yaml:159]: Useless since it will be determined within the composition

File: definition.yaml [hunk 8]
```diff
@@ -0,0 +1,263 @@
...
+                    storageType: 
+                      type: string
+                    writeConnectionSecretToRef:
+                      type: object
+                      properties:
+                        name: 
+                          type: string
+                        namespace:
+                          type: string
+                    providerConfigRef: #.name
```
Comment (braghettos) [inline definition.yaml:232]: Useless since it will be determined within the composition

File: definition.yaml [hunk 9]
```diff
@@ -0,0 +1,263 @@
...
+                  type: object
+                  properties:
+                    role:
+                      type: object
+                      properties:
+                        name:
+                          type: string
+                        assumeRolePolicyDocument:
+                          type: string #it's actually a json
+                        providerConfigRef: #.name
```
Comment (braghettos) [inline definition.yaml:244]: Useless since it will be determined within the composition

File: definition.yaml [hunk 10]
```diff
@@ -0,0 +1,263 @@
...
+                    policyAttachment:
+                      type: object
+                      properties:
+                        name:
+                          type: string
+                        policyArn:
+                          type: string
+                        roleNameRef: #.name
+                          type: string
+                        providerConfigRef: #.name
```
Comment (braghettos) [inline definition.yaml:259]: Useless since it will be determined within the composition

File: claim.yaml [hunk 11]
```diff
@@ -0,0 +1,21 @@
...
+kind: aws-stack
+metadata:
+  name: krateo-aws-stack
+spec:
+  compositionSelector:
+    matchLabels:
+      platform: aws
+  clusterName: #ex clusterRef
+  component_id: #values.component_id
+  networkName: #ex networkRef spec.network.name
```
Comment (braghettos) [inline claim.yaml:12]: I would put it like network.networkName

File: composition.yaml [hunk 12]
```diff
@@ -0,0 +1,418 @@
...
+            - type: string
+              string:
+                fmt: "%s-sn0" 
+        - fromFieldPath: "spec.networkName"
+          toFieldPath: "spec.forProvider.resourcesVpcConfig.subnetIdRefs[1].name" 
+          transforms:
+            - type: string
+              string:
+                fmt: "%s-sn1"
+        - fromFieldPath: "spec.eks.component_id"
```
Comment (braghettos) [inline composition.yaml:41]: Change to spec.name

File: composition.yaml [hunk 13]
```diff
@@ -0,0 +1,418 @@
...
+      patches:
+        - fromFieldPath: "spec.clusterName"
+          toFieldPath: "metadata.name"
+        - fromFieldPath: "spec.networkName"
+          toFieldPath: "spec.forProvider.resourcesVpcConfig.subnetIdRefs[0].name" 
+          transforms:
+            - type: string
+              string:
+                fmt: "%s-sn0" 
+        - fromFieldPath: "spec.networkName"
```
Comment (braghettos) [inline composition.yaml:35]: Change to spec.network.networkName

File: composition.yaml [hunk 14]
```diff
@@ -0,0 +1,418 @@
...
+            - type: string
+              string:
+                fmt: "%s-sn1"
+        - fromFieldPath: "spec.eks.component_id"
+          toFieldPath: "spec.forProvider.roleArnRef"
+          transforms:
+            - type: string
+              string:
+                fmt: "%s-controlplane"
+        - fromFieldPath: "spec.networkName"
```
Comment (braghettos) [inline composition.yaml:47]: Change to spec.network.networkName

File: claim.yaml [hunk 15]
```diff
@@ -0,0 +1,34 @@
+---
+apiVersion: templates.krateo.io/v1alpha1
+kind: aws-stack
+metadata:
+  name: krateo-aws-stack
```
Comment (braghettos) [inline claim.yaml:5]: I would call it 'aws-stack'

**Pull request merged**